src/Tasks/CreateItem.cs

```diff
@@ -142,22 +142,29 @@ private static ITaskItem[] ExpandWildcards(ITaskItem[] expand)
                 {
                     if (FileMatcher.HasWildcards(i.ItemSpec))
                     {
-                        string[] files = FileMatcher.Default.GetFiles(null /* use current directory */, i.ItemSpec);
-                        foreach (string file in files)
+                        try
                         {
-                            TaskItem newItem = new TaskItem(i) { ItemSpec = file };
-
-                            // Compute the RecursiveDir portion.
-                            FileMatcher.Result match = FileMatcher.Default.FileMatch(i.ItemSpec, file);
-                            if (match.isLegalFileSpec && match.isMatch)
+                            string[] files = FileMatcher.Default.GetFiles(null /* use current directory */, i.ItemSpec);
+                            foreach (string file in files)
                             {
-                                if (!string.IsNullOrEmpty(match.wildcardDirectoryPart))
+                                TaskItem newItem = new TaskItem(i) { ItemSpec = file };
+
+                                // Compute the RecursiveDir portion.
+                                FileMatcher.Result match = FileMatcher.Default.FileMatch(i.ItemSpec, file);
+                                if (match.isLegalFileSpec && match.isMatch)
                                 {
-                                    newItem.SetMetadata(FileUtilities.ItemSpecModifiers.RecursiveDir, match.wildcardDirectoryPart);
+                                    if (!string.IsNullOrEmpty(match.wildcardDirectoryPart))
+                                    {
+                                        newItem.SetMetadata(FileUtilities.ItemSpecModifiers.RecursiveDir, match.wildcardDirectoryPart);
+                                    }
                                 }
-                            }
 
-                            expanded.Add(newItem);
+                                expanded.Add(newItem);
+                            }
+                        }
+                        catch (DriveEnumerationWildcardException)
+                        {
+                            throw;
```

Comment:
Test:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="CreateItemTest">
        <CreateItem Include="\**\*.log" />
    </Target>
 </Project>
```
Output:
![image](https://user-images.githubusercontent.com/89041446/144316384-ca22a1d4-bf28-4fd6-88ce-df54a2c48539.png)

Reply:
I believe the custom exception is caught/detected and I throw a new `InvalidProjectFileException` with the custom exception message in TaskBuilder.cs